neo/rawio/baserawio.py

```diff
@@ -233,28 +233,38 @@ def _generate_minimal_annotations(self):
                     d = {}
                     d['name'] = event_channels['name'][c]
                     d['id'] = event_channels['id'][c]
+                    d['file_origin'] = self._source_name()
                     a['blocks'][block_index]['segments'][seg_index]['events'].append(d)
         
         for c in range(signal_channels.size):
             #use for ChannelIndex.annotations
             d = {}
             d['name'] = signal_channels['name'][c]
             d['channel_id'] = signal_channels['id'][c]
+            d['file_origin'] = self._source_name()
             a['signal_channels'].append(d)
 
         for c in range(unit_channels.size):
             #use for Unit.annotations
             d = {}
             d['name'] = unit_channels['name'][c]
             d['id'] = unit_channels['id'][c]
+            d['file_origin'] = self._source_name()
             a['unit_channels'].append(d)
 
+        for c in range(unit_channels.size):
+            #use for ChannelIndex.annotations
+            d = {}
+            d['file_origin'] = self._source_name()
+            a['signal_channels'].append(d)
+
```

Comment:
This lines 255>260 are strange.

You loop over unit_channel.size and you fill annotations for signal_channels.
With this we will have len(raw_annotations['signal_channels'])>signal_channels.size this is not consistent.

My guess is that you do this for having annotations in ChannelIndex that describe a Unit (which is a weakness is actual neo.core and that should change). If so I would prefer no doing this here and remove that lines.
In that case you could use raw_annotations['unit_channels'] to ChannelIndex that are linked to Unit.

Reply:
Yes, that's what I was trying to do. 
So what you mean is adding something like raw_annotations['unit_channels'][channel_id]['channel_index'] as a dictionary for the annotations of ChannelIndex?

Reply:
line 255 to 260 must be removed.

What I meant is in basfromrawio.py:
```python
        elif units_group_mode=='split-all':
            for c in range(len(unit_channels)):
                unit_annotations = self.raw_annotations['unit_channels'][c]
                unit = Unit(**unit_annotations)
                channel_index = ChannelIndex(index=np.array([], dtype='i'),
                                        name='ChannelIndex for Unit')
                channel_index.units.append(unit)
                bl.channel_indexes.append(channel_index)
```
could be change by

```python
        elif units_group_mode=='split-all':
            for c in range(len(unit_channels)):
                unit_annotations = self.raw_annotations['unit_channels'][c]
                unit = Unit(**unit_annotations)
                channel_index = ChannelIndex(index=np.array([], dtype='i'),
                                        name='ChannelIndex for Unit', **unit_annotations)###patch here
                channel_index.units.append(unit)
                bl.channel_indexes.append(channel_index)
```

But finally it is a bad idea. raw_annotations['unit_channels'] must be routed to Unit.annotations only.
ChannelIndex must get raw_annotations['signal_channels'] in case of AnalogSignal and nohting in case of SpikeTrain to avoid confusion (which is already high).